src/ble/BleLayer.cpp

```diff
@@ -104,15 +107,24 @@ class BleEndPointPool
     {
         typedef union
         {
-            uint8_t Data[sizeof(BLEEndPoint)];
-            uint64_t ForceAlignment;
+            uint8_t Data[((sizeof(BLEEndPoint) + BLE_ENDPOINT_ALIGNMENT - 1) / BLE_ENDPOINT_ALIGNMENT) * BLE_ENDPOINT_ALIGNMENT];
+            BLE_ENDPOINT_ALIGNMENT_TYPE ForceAlignment;
         } PoolElement;
 
-        static uint8_t sEndPointPool[sizeof(PoolElement) * BLE_LAYER_NUM_BLE_ENDPOINTS];
+        static union
+        {
+            uint8_t Pool[sizeof(PoolElement) * BLE_LAYER_NUM_BLE_ENDPOINTS];
+            BLE_ENDPOINT_ALIGNMENT_TYPE ForceAlignment;
+        } sEndPointPool;
```

Comment:
This is unnecessarily complex.   First off, sizeof() a type always includes the padding necessary to ensure that every element in an array of such types is aligned, provided that the array itself is aligned.  So there's no need to round the sizeof an element up to the sizeof its alignment type.

Additionally (and really for the same reason) one only needs to force the alignment of the entire array, not each element.

Thus the above unions can be simplified to just:

```
static union
{
    uint8_t Pool[sizeof(BLEEndPoint) * BLE_LAYER_NUM_BLE_ENDPOINTS];
    BLE_ENDPOINT_ALIGNMENT_TYPE ForceAlignment;
} sEndPointPool;
```

Finally, don't use #defines for types, use typedefs.  In this case, the type being defined is intimately related to the definition of BLEEndPoint.  So it is customary to declare the alignment type as a public member of the to-be-aligned class.  E.g.:

```
class NL_DLL_EXPORT BLEEndPoint : public BleLayerObject
{
public:
    typedef uint64_t AlignT;
    ...
}
```

which would reduce the pool definition to:

```
static union
{
    uint8_t Pool[sizeof(BLEEndPoint) * BLE_LAYER_NUM_BLE_ENDPOINTS];
    BLEEndPoint::AlignT ForceAlignment;
} sEndPointPool;
```

Reply:
The first block of code (reproduced below to be specific) was my first approach exactly, but only the first element was aligned, the second element and beyond were not aligned when the compiler optimizations were included. It works fine without the optimizations, though.

```c++
static union
{
    uint8_t Pool[sizeof(BLEEndPoint) * BLE_LAYER_NUM_BLE_ENDPOINTS];
    BLE_ENDPOINT_ALIGNMENT_TYPE ForceAlignment;
} sEndPointPool;
```

Good point on the typedef and keeping it in the class, not sure what I was thinking with that.